addon/services/router-scroll.js

```diff
@@ -29,6 +80,111 @@ class RouterScroll extends Service {
         x: 0, y: 0
       }
     });
+
+    // https://github.com/ember-app-scheduler/ember-app-scheduler/pull/773
+    setupRouter(this.router);
+
+    addListener(this.router, 'routeWillChange', this._routeWillChange);
+    addListener(this.router, 'routeDidChange', this._routeDidChange);
+  }
+
+  destroy() {
+    reset();
+
+    removeListener(this.router, 'routeWillChange', this._routeWillChange);
+    removeListener(this.router, 'routeDidChange', this._routeDidChange);
+
+    if (requestId) {
+      window.cancelAnimationFrame(requestId);
+    }
+
+    if (callbackRequestId) {
+      window.cancelAnimationFrame(callbackRequestId);
+    }
+
+    super.destroy(...arguments);
+  }
+
+  /**
+   * Updates the scroll position
+   * it will be a single transition
+   * @method updateScrollPosition
+   * @param {transition|transition[]} transition If before Ember 3.6, this will be an array of transitions, otherwise
+   */
+  updateScrollPosition(transition) {
+    const url = get(this, 'currentURL');
+    const hashElement = url ? document.getElementById(url.split('#').pop()) : null;
+
+    if (get(this, 'isFirstLoad')) {
+      this.unsetFirstLoad();
+    }
+
+    let scrollPosition;
+
+    if (url && url.indexOf('#') > -1 && hashElement) {
+      scrollPosition = { x: hashElement.offsetLeft, y: hashElement.offsetTop };
+    } else {
+      scrollPosition = get(this, 'position');
+    }
+
+    let preserveScrollPosition = (get(transition, 'router.currentRouteInfos') || []).some((routeInfo) => get(routeInfo, 'route.controller.preserveScrollPosition'));
+
+    // If `preserveScrollPosition` was not set on the controller, attempt fallback to `preserveScrollPosition` which was set on the router service.
+    if (!preserveScrollPosition) {
+      preserveScrollPosition = get(this, 'preserveScrollPosition')
+    }
+
+    if (!preserveScrollPosition) {
+      const scrollElement = get(this, 'scrollElement');
+      const targetElement = get(this, 'targetElement');
```

Comment:
Do you need the `get` or could you use `const { scrollElement, targetElement } = this;` ?